OpenSim/Simulation/StatesTrajectory.cpp

```diff
@@ -143,16 +143,21 @@ TimeSeriesTable StatesTrajectory::exportToTable(const Model& model,
             requestedStateVars;
     table.setColumnLabels(stateVars);
     size_t numDepColumns = stateVars.size();
-
+    
     // Fill up the table with the data.
     for (size_t itime = 0; itime < getSize(); ++itime) {
         const auto& state = get(itime);
         TimeSeriesTable::RowVector row(static_cast<int>(numDepColumns));
 
         // Get each state variable's value.
-        for (unsigned icol = 0; icol < numDepColumns; ++icol) {
-            row[static_cast<int>(icol)] = 
-                model.getStateVariableValue(state, stateVars[icol]);
+        if (requestedStateVars.empty()) {
+            // This is *much* faster than getting the values one-by-one.
```

Comment:
> I suppose I could do some testing to determine the number of state variables at which to switch schemes.

Might vary depending on the model, etc. It would be slick if StatesTrajectory ran internal tests (e.g., during initialization) to determine whether to take an "additive" or "subtractive" approach, but I think there are more pressing issues for 4.0. Really, I was just imagining a question on the Forum asking how to speed up StatesTrajectory, and the nonintuitive reply to export all state variables instead of a subset.